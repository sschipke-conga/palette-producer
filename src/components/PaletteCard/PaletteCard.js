import './PaletteCard.scss';
import React from 'react';
import PropTypes from 'prop-types';

export const PaletteCard = ({ colors }) => {

  const displayColors = colors.map(color => {
    const colorStyles = {
      background: color,
    }
    return (
      <div className='colorblocks' style={colorStyles}>
      </div>
    )
  })

  return (
    <div className='PaletteCard'>
      {displayColors}
    </div >
  )
}

export default PaletteCard;

PaletteCard.propTypes = {

}
