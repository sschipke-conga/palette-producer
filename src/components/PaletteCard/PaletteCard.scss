.PaletteCard-div {
  margin-left: 20px;
}

.PaletteCard {
  border: 2px black solid;
  display: flex;
  flex: 0 0 auto;
  height: 150px;
  width: 200px;
}

.paletteCard-header {
  margin-bottom: 5px;
}

.colors-div {
  display: flex;
  height: 150px;
  width: 150px;
}

.colorblocks {
  height: 100%;
  width: 20%;
}

.palette-buttons-div {
  display: flex;
  justify-content: space-between;
  padding: 0 15px;

  .palette-button {
  border: none;
  border-radius: 5px;
  cursor: pointer;
  height: 50px;
  margin-top: 10px;
  padding: 0;
  opacity: 0.6;
  width: 50px;

  &:hover {
    background: #FDF7DB;
    color: black;
    opacity: 1;
  }
  &:disabled {
    cursor: not-allowed;
    opacity: 20%;
    &:hover {
      opacity: 5%;
    }
  }
    .delete-icon, .edit-icon {
      height: 100%;
      width: 100%;
      &:disabled {
      cursor: not-allowed;
      }
    }
  }
}