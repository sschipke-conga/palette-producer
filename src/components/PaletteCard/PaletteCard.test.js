import React from "react";
import PaletteCard from "./PaletteCard";
import { shallow } from "enzyme";

describe('PaletteCard', () => {
  const mockProject={
    id: 1,
    name: 'Greg'
  }
  const mockPalette = {
    id: 1,
    name: "Palette",
    project_id: 1,
    color1: "#FFFFFF",
    color2: "#FFFFFF",
    color3: "#000000",
    color4: "#FFFFFF",
    color5: "#FFFFFF"
  };
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<PaletteCard 
    select={jest.fn()}
    project={mockProject}
    palette={mockPalette}
    removePalette={jest.fn()}
    />)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})