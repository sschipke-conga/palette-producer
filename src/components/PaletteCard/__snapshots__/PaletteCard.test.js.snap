// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteCard should match the snapshot 1`] = `
<div
  className="PaletteCard-div"
>
  <h4
    className="paletteCard-header"
  >
    Palette
  </h4>
  <div
    className="PaletteCard"
    id={1}
    name="Palette"
    onClick={[Function]}
  >
    <div
      className="colorblocks"
      key="0"
      style={
        Object {
          "background": "#FFFFFF",
        }
      }
    />
    <div
      className="colorblocks"
      key="1"
      style={
        Object {
          "background": "#FFFFFF",
        }
      }
    />
    <div
      className="colorblocks"
      key="2"
      style={
        Object {
          "background": "#000000",
        }
      }
    />
    <div
      className="colorblocks"
      key="3"
      style={
        Object {
          "background": "#FFFFFF",
        }
      }
    />
    <div
      className="colorblocks"
      key="4"
      style={
        Object {
          "background": "#FFFFFF",
        }
      }
    />
  </div>
  <button
    id={1}
    onClick={[MockFunction]}
    type="button"
  >
    X
  </button>
</div>
`;
